recipes/libmysqlclient/all/conanfile.py

```diff
@@ -0,0 +1,106 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+import glob
+
+
+
+class libMysqlClientCConan(ConanFile):
+    name = "libmysqlclient"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A MySQL client library for C development."
+    topics = ("conan", "mysql", "sql", "connector", "database")
+    homepage = "https://dev.mysql.com/downloads/mysql/"
+    license = "GPL-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_ssl": [True, False], "with_zlib": [True, False]}
+    default_options = {"shared": True, "fPIC": True, "with_ssl": True, "with_zlib": True}
+    _source_subfolder = "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_ssl:
+            self.requires.add("openssl/1.1.1d")
+
+        if self.options.with_zlib:
+            self.requires.add("zlib/1.2.11")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        archive_name = "mysql-" + self.version
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+        for lib in ["icu", "libevent", "re2", "rapidjson", "lz4", "protobuf", "libedit"]:
+            tools.rmdir(os.path.join(self._source_subfolder, "extra", lib))
+        for folder in ['client', 'man', 'mysql-test']:
+            tools.rmdir(os.path.join(self._source_subfolder, folder))
+        tools.rmdir(os.path.join(self._source_subfolder, "storage", "ndb"))
+
+    def _patch(self):
+        tools.patch(**self.conan_data["patches"][self.version])
+        sources_cmake = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        sources_cmake_orig = os.path.join(self._source_subfolder, "CMakeListsOriginal.txt")
+        os.rename(sources_cmake, sources_cmake_orig)
+        os.rename("CMakeLists.txt", sources_cmake)
+        if self.settings.os == "Macos":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "libmysql", "CMakeLists.txt"), "COMMAND $<TARGET_FILE:libmysql_api_test>", "COMMAND DYLD_LIBRARY_PATH=%s $<TARGET_FILE:libmysql_api_test>" % os.path.join(self.build_folder, "library_output_directory"))
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self.options.shared:
+            raise ConanInvalidConfiguration("libmysqlclient cannot be built as static library")
+        if self.settings.compiler == "Visual Studio"
```

Comment:
you have a missing `:` here:
```suggestion
        if self.settings.compiler == "Visual Studio":
```

Reply:
That's embarrassing. thanks!